pulp_container/app/replica.py

```diff
@@ -0,0 +1,53 @@
+from pulpcore.plugin.replica import Replicator
+from pulpcore.plugin.util import get_url
+
+from pulp_glue.container.context import (
+    PulpContainerDistributionContext,
+    PulpContainerRepositoryContext,
+)
+
+from pulp_container.app.models import ContainerDistribution, ContainerRemote, ContainerRepository
+from pulp_container.app.tasks import synchronize as container_synchronize
+
+
+class ContainerReplicator(Replicator):
+    distribution_ctx_cls = PulpContainerDistributionContext
+    repository_ctx_cls = PulpContainerRepositoryContext
+    remote_model_cls = ContainerRemote
+    repository_model_cls = ContainerRepository
+    distribution_model_cls = ContainerDistribution
+    distribution_serializer_name = "ContainerDistributionSerializer"
+    repository_serializer_name = "ContainerRepositorySerializer"
+    remote_serializer_name = "ContainerRemoteSerializer"
+    app_label = "container"
+    sync_task = container_synchronize
+
+    def sync_params(self, repository, remote):
+        """Returns a dictionary where key is a parameter for the sync task."""
+        return dict(
+            remote_pk=str(remote.pk),
+            repository_pk=str(repository.pk),
+            signed_only=False,
+            mirror=True,
+        )
+
+    def url(self, upstream_distribution):
+        return self.pulp_ctx._api_kwargs["base_url"]
+
+    def remote_extra_fields(self, upstream_distribution):
+        upstream_name = upstream_distribution["registry_path"].split("/")[1]
```

Comment:
registry_path is not available on the model, can you use base_path?

Reply:
@ipanova how is it not available? ContainerDistributionSerializer has it defined, and the replication had been tested, it works

Reply:
`upstream_distribution` here is an instance of distribution model?

Reply:
yep, I'm pretty sure

Reply:
well in that case distribution model does not have registry_path as a field on the model https://github.com/pulp/pulp_container/blob/main/pulp_container/app/models.py#L712

Reply:
I believe it is serialized and just passed as a dictionary, sorry for misinterpreting the previous question. Anyway, the code has been already used/tested

https://github.com/pulp/pulp_container/blob/da8a70f6306b68bc5bdbd66dd0879628a30e1b43/pulp_container/app/serializers.py#L404